ray-operator/controllers/ray/common/service.go

```diff
@@ -166,6 +166,13 @@ func BuildServeServiceForRayCluster(ctx context.Context, rayCluster rayv1.RayClu
 	return BuildServeService(ctx, rayv1.RayService{}, rayCluster, false)
 }
 
+func ServeServiceNameForRayService(rayService *rayv1.RayService) string {
+	if rayService.Spec.ServeService != nil && rayService.Spec.ServeService.Name != "" {
+		return rayService.Spec.ServeService.Name
+	}
+	return utils.GenerateServeServiceName(rayService.Name)
```

Comment:
Note that users can use their custom-defined serve service; it is not always the case that the service name is equal to utils.GenerateServeServiceName(rayService.Name). See this sample YAML file: https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-service.custom-serve-service.yaml

Reply:
Thanks! And that is why we check the condition `rayService.Spec.ServeService != nil` first.